airsonic-main/src/main/java/org/airsonic/player/service/AudioScrobblerService.java

```diff
@@ -18,309 +18,60 @@
-        }
-
-        HttpPost request = new HttpPost(url);
-        request.setEntity(new UrlEncodedFormEntity(params, StringUtil.ENCODING_UTF8));
-        request.setConfig(requestConfig);
-        return executeRequest(request);
-    }
-
-    private String[] executeRequest(HttpUriRequest request) throws IOException {
-
-        try (CloseableHttpClient client = HttpClients.createDefault()) {
-            ResponseHandler<String> responseHandler = new BasicResponseHandler();
-            String response = client.execute(request, responseHandler);
-            return response.split("\\n");
-
-        }
-    }
-
-    public void setSettingsService(SettingsService settingsService) {
-        this.settingsService = settingsService;
-    }
-
-    private class RegistrationThread extends Thread {
-        private RegistrationThread() {
-            super("AudioScrobbler Registration");
-        }
-
-        @Override
-        public void run() {
-            while (true) {
-                RegistrationData registrationData = null;
-                try {
-                    registrationData = queue.take();
-                    scrobble(registrationData);
-                } catch (IOException x) {
-                    handleNetworkError(registrationData, x);
-                } catch (Exception x) {
-                    LOG.warn("Error in Last.fm registration.", x);
-                }
+        UserSettings userSettings = settingsService.getUserSettings(username);
+        if (userSettings.isLastFmEnabled() && userSettings.getLastFmUsername() != null && userSettings.getLastFmPassword() != null) {
+            if (lastFMScrobbler == null) {
+                lastFMScrobbler = new LastFMScrobbler();
             }
+            lastFMScrobbler.register(mediaFile, userSettings.getLastFmUsername(), userSettings.getLastFmPassword(), submission, time);
         }
 
-        private void handleNetworkError(RegistrationData registrationData, IOException x) {
-            try {
-                queue.put(registrationData);
-                LOG.info("Last.fm registration for " + registrationData.title +
-                         " encountered network error.  Will try again later. In queue: " + queue.size(), x);
-            } catch (InterruptedException e) {
-                LOG.error("Failed to reschedule Last.fm registration for " + registrationData.title, e);
-            }
-            try {
-                sleep(60L * 1000L);  // Wait 60 seconds.
-            } catch (InterruptedException e) {
-                LOG.error("Failed to sleep after Last.fm registration failure for " + registrationData.title, e);
+        if (userSettings.isListenBrainzEnabled() && userSettings.getListenBrainzToken() != null) {
+            if (listenBrainzScrobbler == null) {
+                listenBrainzScrobbler = new ListenBrainzScrobbler();
             }
+            listenBrainzScrobbler.register(mediaFile, userSettings.getListenBrainzToken(), submission, time);
```

Comment:
Same comment about token in the ctor. Make an interface with a `register(mediaFile, submission, time)` method?

Reply:
Same comment above.